File: examples/supplychain-app/fabric/chaincode_rest_server/chaincode/common/ContainerRequest.go
package common

// The ContainerRequest models a request body for container creation in a supply chain
type ContainerRequest struct {
	ID           string                 `json:"trackingID"`
	Health       string                 `json:"health"`
	Metadata     map[string]interface{} `json:"misc"`
	Location     string                 `json:"lastScannedAt"`
	Participants []string               `json:"counterparties"`
}

File: examples/supplychain-app/fabric/chaincode_rest_server/rest-server/src/models/ContainerRequest.ts
import { deserialize } from 'class-transformer';
import { IsDefined, IsNotEmpty, IsUUID } from 'class-validator';
import { Validateable } from 'mixins/Validateable';

export class ContainerRequest extends Validateable {
  static fromJSON(json: string): ContainerRequest {
    return deserialize(ContainerRequest, json);
  }

  @IsNotEmpty()
  @IsDefined()
  health: string;

  @IsNotEmpty()
  @IsDefined()
  misc: { [key: string]: any };

  @IsNotEmpty()
  @IsDefined()
  @IsUUID()
  trackingID: string;

  @IsNotEmpty()
  @IsDefined()
  counterparties: string[];

  lastScannedAt: string;
}
